### PR TITLE
Configurable parameter for headers-chain syncing status

### PR DESCRIPTION
**Pull request by kushti**
A node is doing headers-chain sync first. It is considered that the headers-chain is synced if currently observed tip is no older than some some parameter (previously hardcoded, 100 blocks ~= 200 minutes on average). After syncing headers-chain the node is deciding which full blocks to download. E.g. in the pruned regime, it downloads only certain suffix of the blockchain (determined by blocksToKeep parameter).

In this PR, this difference to decide syncing status (for the headers-chain) is made configurable.

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -61,6 +61,11 @@ ergo {
 
     # Minimal fee amount of transactions mempool accepts
     minimalFeeAmount = 1000000
+
+    # A node is considering that the chain is synced if sees a block header with timestamp no more
+    # than headerChainTimeDiff blocks on average from future
```
Comment (aslesarenko) [inline src/main/resources/reference.conf]: Please reformulate is simplier way, maybe with example.

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala [hunk 2]
```diff
@@ -54,10 +56,13 @@ trait ToDownloadProcessor extends BasicReaders with ScorexLogging {
     }
 
     bestFullBlockOpt match {
+        //do not download full blocks if no headers-chain synced yet or SPV mode
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala]: let's move comments inside the body of the corresponding branch.

File: src/test/scala/org/ergoplatform/settings/ErgoSettingsSpecification.scala [hunk 3]
```diff
@@ -17,21 +17,21 @@ class ErgoSettingsSpecification extends ErgoPropertyTest {
     settings.nodeSettings shouldBe NodeConfigurationSettings(StateType.Utxo, verifyTransactions = true, 1000,
       poPoWBootstrap = false, 10, mining = false, Constants.DefaultComplexityLimit, 1.second, useExternalMiner = false, miningPubKeyHex = None,
       offlineGeneration = false, keepVersions = 200, mempoolCapacity = 100000, blacklistCapacity = 100000,
-      mempoolCleanupDuration = 10.seconds, rebroadcastCount = 3, minimalFeeAmount = 0)
+      mempoolCleanupDuration = 10.seconds, rebroadcastCount = 3, minimalFeeAmount = 0, headerChainTimeDiff = 100)
```
Comment (aslesarenko) [inline src/test/scala/org/ergoplatform/settings/ErgoSettingsSpecification.scala]: we are not testing that parameter can be changed. Maybe clone the test cases and use some other value?

Comment (kushti) [reply src/test/scala/org/ergoplatform/settings/ErgoSettingsSpecification.scala]: In these tests we're reading settings from a file and check that expected values being read from it. Values can not be changed. We can check another branch in a way like: 

(settings.nodeSettings == NodeConfigurationSettings(StateType.Utxo, verifyTransactions = true, 1000,
      poPoWBootstrap = false, 10, mining = false, Constants.DefaultComplexityLimit, 1.second, useExternalMiner = false, miningPubKeyHex = None,
      offlineGeneration = false, keepVersions = 200, mempoolCapacity = 100000, blacklistCapacity = 100000,
      mempoolCleanupDuration = 10.seconds, rebroadcastCount = 3, minimalFeeAmount = 0, headerChainTimeDiff = 10)) shouldBe false

but I'm not sure we're checking something new here

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala (hunk 2 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala]: done

File: src/main/resources/reference.conf (hunk 1 above)
Comment (kushti) [reply src/main/resources/reference.conf]: Improved description.

**Comment by kushti** [comment]
@knizhnik @aslesarenko thanks for the reviews!

renaming done: "headerChainTimeDiff" -> "headerChainDiff"

comments improved.

Please make another round!

**Pull request merged**